src/main/webapp/script.js

```diff
@@ -18,21 +18,196 @@ var arrayOfProducts = Array();
+           let prodType = arrayOfProducts[i].product_type;
+           if(prodType == valName){
+                str += prodBrand + ": " + prodName + "<br>";
            }
         }
+        document.getElementById("eyeliner_").innerHTML = str;
+    }else{
+        document.getElementById("eyeliner_").innerHTML = " ";
     }
 } 
-async function doStuff(){
-    await fetchDataAsync();
-    filtering();
-}
\ No newline at end of file
+
+function filteringEyeshadow(){
+    let str = " ";
+    var checkBox = document.getElementById("eyeshadow");
+    if(checkBox.checked == true){
+        for(i = 0; i <  arrayOfProducts.length; i++){
+           let prodBrand = arrayOfProducts[i].brand;
+           let valName = document.getElementById("eyeshadow").value;
+           let prodName = arrayOfProducts[i].name;
+           let prodType = arrayOfProducts[i].product_type;
+           if(prodType == valName){
+                 str += prodBrand + ": " + prodName + "<br>";
+           }
+        }
+        document.getElementById("eyeshadow_").innerHTML = str;
+    }else{
+        document.getElementById("eyeshadow_").innerHTML = " ";
+    }
+} 
+
+function filteringLipliner(){
+    let str = " ";
+    var checkBox = document.getElementById("lipliner");
+    if(checkBox.checked == true){
+        for(i = 0; i <  arrayOfProducts.length; i++){
+           let prodBrand = arrayOfProducts[i].brand;
+           let valName = document.getElementById("lipliner").value;
+           let prodName = arrayOfProducts[i].name;
+           let prodType = arrayOfProducts[i].product_type;
+           if(prodType == valName){
+                str += prodBrand + ": " + prodName + "<br>";
+           }
+        }
+        document.getElementById("lipliner_").innerHTML = str;
+    }else{
+        document.getElementById("lipliner_").innerHTML = " ";
+    }
+} 
+
+function filteringMascara(){
+    var checkBox = document.getElementById("mascara");
+    if(checkBox.checked == true){
+        for(i = 0; i <  arrayOfProducts.length; i++){
+           let prodBrand = arrayOfProducts[i].brand;
+           let valName = document.getElementById("mascara").value;
+           let prodName = arrayOfProducts[i].name;
+           let prodType = arrayOfProducts[i].product_type;
+           if(prodType == valName){
+               str += prodBrand + ": " + prodName + "<br>";
+           }
+        }
+        document.getElementById("mascara_").innerHTML = str;
+    }else{
+        document.getElementById("mascara_").innerHTML = " ";
+    }
+} 
+
+//filtering by tags
+function gluten(){
+    let str = " ";
+    var checkBox = document.getElementById("gluten free");
+    if(checkBox.checked == true){
+        for(i = 0; i <  arrayOfProducts.length; i++){
+           let prodTags = arrayOfProducts[i].tag_list;
+           let prodBrand = arrayOfProducts[i].brand;
+           let valName = document.getElementById("gluten free").value;
+           let prodName = arrayOfProducts[i].name;
```

Comment:
There is a way to do the filtering without all of these separate tags in one function. I would not suggest doing it in this way.